vault-core/src/main/java/org/apache/jackrabbit/vault/util/xml/serialize/XMLSerializer.java

```diff
@@ -1100,6 +1109,73 @@ private void printNamespaceAttr(String prefix, String uri) throws IOException {
         _printer.printText('"');
     }
 
+    private List<Node> sortAttributes(NamedNodeMap attributeList) {
```

Comment:
I dont like this change. if there is no comparator defined, you add the overhead of copying all attributes into a new list.

Reply:
According to NamedNodeMap's javadoc 

> Objects contained in an object implementing NamedNodeMap may also be accessed by an ordinal index, but this is simply to allow convenient enumeration of the contents of a NamedNodeMap, and does not imply that the DOM specifies an order to these Nodes

the attributes in the NamedNodeMap don't imply an order. Accessing using the index is just for convenience. 

As the usage of the attributes is sequential access only, I changed that to use an Iterable around the NamedNodeMap when no Comparator is given, and the List as before otherwise.